WireAnalytics/Sources/WireDatadogTracker/WireDatadogTracker.swift

```diff
@@ -0,0 +1,120 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CryptoKit
+import DatadogCore
+import DatadogCrashReporting
+import DatadogLogs
+import DatadogRUM
+import DatadogTrace
+import UIKit
+
+public final class WireDatadogTracker {
+
+    private let applicationID: String
+    private let logLevel: LogLevel = .debug
+    private let userIdentifier: String
+
+    public var datadogUserIdentifier: String? { userIdentifier }
+
+    public private(set) var logger: (any DatadogLogs.LoggerProtocol)?
+
+    public init(
+        appID: String,
+        clientToken: String,
+        identifierForVendor: UUID?,
+        environmentName: String
+    ) {
+        applicationID = appID
+
+        if let identifierForVendor {
+            userIdentifier = Self.hashedDatadogUserIdentifier(identifierForVendor)
+        } else {
+            userIdentifier = "none"
+        }
```

Comment:
I didn't catch that the first time... I don't see how the app could run without user unlocked the device, or maybe the NSE, in that case I don't think it can even go further as all files should be restricted access

Reply:
Isn't that the case when you ran out of battery, turning on the phone again and receiving missing messages?
I would consider it quite common.

Reply:
let's keep it for now, as this behaviour existed before